test/test_jit.py

```diff
@@ -5620,6 +5620,120 @@ def backward_hook(module, grad_input, grad_output):
         with self.assertRaisesRegex(Exception, "backward hooks assigned"):
             torch.jit.trace(n, (torch.tensor(1.0),))
 
+    def test_scripting_hooks(self):
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+
+            def forward(self, x):
+                return x + x
+
+        def forward_pre_hook(self, inputs):
+            # type: (Tuple[Tensor,]) -> Tuple[Tensor,]
+            return (inputs[0] + 10,)
+
+        def forward_hook(self, inputs, outputs):
+            return outputs + 30
+
+        n = Net()
+        n.register_forward_pre_hook(forward_pre_hook)
+        scripted_n = torch.jit.script(n)
```

Comment:
Hooks should work with our existing test checking stuff, so you could replace this (and all the other `torch.jit.script`...`assertEqual`s) and the `assertEqual`s below with `self.checkModule(n, [t])` (that will run it in eager and script mode and compare the results)